initial changes to documentation menu structure

eldonquijote opened the pull request:
## Purpose / why

Evolution of Islandora documentation structure, facilitate gap identification

## What changes were made?

- changed documentation structure by moving and renaming menu items
- added 'permalink' feature to mkdocs config to add link anchors to headings
- added placeholder page so we are able to add menu entries that are still awaiting documentation to be written

## Verification

check for unclear page titles, add placeholder entries

## Interested Parties

> @manez @rosiel @dannylamb Jeff?
---

## Checklist

> __Pull-request reviewer__ should ensure the following

* [ ] Does this PR link to related [issues](https://github.com/Islandora/documentation/issues/)?
* [ ] Does the proposed documentation align with the [Islandora Documentation Style Guide](https://islandora.github.io/documentation/contributing/docs_style_guide/)?
* [ ] Are the changes accurate, useful, free of typos, etc?

> __Person merging__ should ensure the following
* [ ] Does mkdocs still build successfully? (This is indicated by TravisCI passing. To test locally, and see warnings, see [How To Build Documentation](https://islandora.github.io/documentation/technical-documentation/docs-build/).)
* [ ] If pages are renamed or removed, have all internal links to those pages been fixed?
* [ ] If pages are added, have they been linked to or placed in the menu?
* [ ] Did the PR receive at least one approval from a committer, and all issues raised have been addressed?

eldonquijote committed:
initial changes to documentation menu structure

manez committed:
Update placeholder.md

manez commented:
Er.. I tried to make a pull request against your branch but instead I seem to have pushed the change. Please feel free to revert if you don't like the suggestion to add the DIG to the placeholder page. It was meant as *just* a suggestion 😅 

I did a test build with these changes and everything looks good! Since I now have a commit on this one, we'll need some one else to merge.

eldonquijote committed:
Update mkdocs.yml

Renamed menu entry "Add a New User" to "Create and Manage User Accounts" to be in line with Melissa's earlier PR

eldonquijote committed:
Merge branch 'main' into documentation-structure-evolution

manez commented:
Tagging @jrubin6 since he is name-dropped in the PR too :)

rosiel commented on mkdocs.yml:
```diff
@@ -116,7 +201,8 @@ nav:
         - 'Resizing a VM': 'technical-documentation/resizing_vm.md'
       - 'Checking Coding Standards': 'technical-documentation/checking-coding-standards.md'
       - 'Contributing Workflow': 'contributing/contributing-workflow.md'
-      - 'Editing Documentation': 'contributing/editing-docs.md'
```
This is @manez 's new tutorial for people who haven't made a PR before, can we keep it in?

eldonquijote replied:
Good eye, I must have lost that one with all the reshuffling. I'll push an update in a bit

eldonquijote committed:
adding missing menu entry on editing docs

eldonquijote committed:
Merge branch 'documentation-structure-evolution' of https://github.com/eldonquijote/documentation into documentation-structure-evolution

rosiel committed:
Merge branch 'main' into documentation-structure-evolution

rosiel reviewed (approved):
Looks good! @Islandora/8-x-committers Can someone swoop in for the merge, since I technically am on the commit list for this PR?
